api/v1alpha1/addon_types.go

```diff
@@ -134,40 +142,158 @@ func (ClusterAutoscaler) VariableSchema() clusterv1.VariableSchema {
+				},
+				"volumeBindingMode": {
+					Type: "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(supportedBindingModes...),
+				},
+			},
+		},
+	}
+}
+
+func (CSIProvider) VariableSchema() clusterv1.VariableSchema {
+	supportedCSIProviders := []string{CSIProviderAWSEBS, CSIProviderNutanix}
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"name": {
+					Description: "Name of the CSI Provider",
+					Type:        "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(
+						supportedCSIProviders...),
+				},
+				"strategy": {
+					Description: "Addon strategy used to deploy the CSI provider to the workload cluster",
+					Type:        "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(
+						AddonStrategyClusterResourceSet,
+						AddonStrategyHelmAddon,
+					),
+				},
+				"credentials": {
+					Type:        "object",
+					Description: "The reference to any secret used by the CSI Provider.",
+					Properties: map[string]clusterv1.JSONSchemaProps{
+						"name": {
+							Type: "string",
+						},
+						"namespace": {
+							Type: "string",
+						},
+					},
+				},
+				"storageClassConfig": {
+					Type: "array",
+					Items: &clusterv1.JSONSchemaProps{
+						Type:       "object",
+						Properties: StorageClassConfig{}.VariableSchema().OpenAPIV3Schema.Properties,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (DefaultStorage) VariableSchema() clusterv1.VariableSchema {
+	supportedCSIProviders := []string{CSIProviderAWSEBS, CSIProviderNutanix}
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type:        "object",
+			Description: "A tuple of provider name and storage class ",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"providerName": {
+					Type:        "string",
+					Description: "Name of the CSI Provider for the default storage class",
+					Enum: variables.MustMarshalValuesToEnumJSON(
+						supportedCSIProviders...),
+				},
+				"storageClassConfigName": {
+					Type:        "string",
+					Description: "Name of storage class config in any of the provider objects",
+				},
+			},
+		},
+	}
 }
 
 func (CSIProviders) VariableSchema() clusterv1.VariableSchema {
-	supportedCSIProviders := []string{CSIProviderAWSEBS}
 	return clusterv1.VariableSchema{
 		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
 			Type: "object",
 			Properties: map[string]clusterv1.JSONSchemaProps{
 				"providers": {
 					Type: "array",
 					Items: &clusterv1.JSONSchemaProps{
-						Type: "object",
-						Properties: map[string]clusterv1.JSONSchemaProps{
-							"name": {
-								Type: "string",
-								Enum: variables.MustMarshalValuesToEnumJSON(
-									supportedCSIProviders...),
-							},
-						},
+						Type:       "object",
+						Properties: CSIProvider{}.VariableSchema().OpenAPIV3Schema.Properties,
```

Comment:
Use the OpenAPI schema from `CSIProvider` as the schema for items

```suggestion
					Items: ptr.To(CSIProvider{}.VariableSchema().OpenAPIV3Schema},
```